src/task/process.ts

```diff
@@ -0,0 +1,65 @@
+import { Readable, Transform } from "stream";
+import { Result } from "../activity";
+
+const DEFAULTS = {
+  exitWaitingTimeout: 20_000,
+};
+/**
+ * RemoteProcess class representing the process spawned on the provider by {@link WorkContext.spawn}
+ */
+export class RemoteProcess {
+  /**
+   * Stream connected to stdout from provider process
+   */
+  readonly stdout: Readable;
+  /**
+   * Stream connected to stderr from provider process
+   */
+  readonly stderr: Readable;
+  private lastResult?: Result;
+  private streamError?: Error;
+  constructor(private streamOfActivityResults: Readable) {
+    this.streamOfActivityResults.on("data", (data) => (this.lastResult = data));
+    this.streamOfActivityResults.on("error", (error) => (this.streamError = error));
+    const { stdout, stderr } = this.transformResultsStream();
+    this.stdout = stdout;
+    this.stderr = stderr;
+  }
+
+  /**
+   * Waits for the process to complete and returns the last part of the command's results as a {@link Result} object
+   * @param timeout - maximum waiting time for the final result (default: 20 sec)
+   */
+  waitForExit(timeout?: number): Promise<Result> {
+    return new Promise((res, rej) => {
```

Comment:
can we use full names for resolve and reject? I think it gives a more meaningful name for the variables